platform/broadcom/sonic-platform-modules-dell/s6000/sonic_platform/psu.py

```diff
@@ -0,0 +1,227 @@
+        """
+        status = False
+        psu_status = self._get_cpld_register(self.psu_status_reg)
+        if (psu_status != 'ERR'):
+            psu_status = (int(psu_status, 16) >> ((2 - self.index) * 4)) & 0xF
+            if (~psu_status & 0b1000) and (~psu_status & 0b0100):
+                status = True
+
+        return status
+
+    def get_voltage(self):
+        """
+        Retrieves current PSU voltage output
+
+        Returns:
+            A float number, the output voltage in volts,
+            e.g. 12.1
+        """
+        psu_voltage = self._get_i2c_register(self.psu_voltage_reg)
+        if (psu_voltage != 'ERR') and self.get_status():
+            # Converting the value returned by driver which is in
+            # millivolts to volts
+            psu_voltage = float(psu_voltage) / 1000
+        else:
+            psu_voltage = 0.0
+
+        return psu_voltage
+
+    def get_current(self):
+        """
+        Retrieves present electric current supplied by PSU
+
+        Returns:
+            A float number, electric current in amperes,
+            e.g. 15.4
+        """
+        psu_current = self._get_i2c_register(self.psu_current_reg)
+        if (psu_current != 'ERR') and self.get_status():
+            # Converting the value returned by driver which is in
+            # milliamperes to amperes
+            psu_current = float(psu_current) / 1000
+        else:
+            psu_current = 0.0
+
+        return psu_current
+
+    def get_power(self):
+        """
+        Retrieves current energy supplied by PSU
+
+        Returns:
+            A float number, the power in watts,
+            e.g. 302.6
+        """
+        psu_power = self._get_i2c_register(self.psu_power_reg)
+        if (psu_power != 'ERR') and self.get_status():
+            # Converting the value returned by driver which is in
+            # microwatts to watts
+            psu_power = float(psu_power) / 1000000
+        else:
+            psu_power = 0.0
+
+        return psu_power
+
+    def get_powergood_status(self):
+        """
+        Retrieves the powergood status of PSU
+        Returns:
+            A boolean, True if PSU has stablized its output voltages and
+            passed all its internal self-tests, False if not.
+        """
+        status = False
+        psu_status = self._get_cpld_register(self.psu_status_reg)
+        if (psu_status != 'ERR'):
+            psu_status = (int(psu_status, 16) >> ((2 - self.index) * 4)) & 0xF
+            if (psu_status == 0x2):
+                status = True
+
+        return status
+
+    def set_status_led(self):
+        """
+        Sets the state of the PSU status LED
+        Args:
+            color: A string representing the color with which to set the
+                   PSU status LED
+        Returns:
+            bool: True if status LED state is set successfully, False if
+                  not
+        """
+        # Hardware not supported
+        return False
```

Comment:
The S6000 has status LEDs on each PSU and also one power status LED on the front panel. How are these controlled?

Reply:
In S6000, the firmware running in the PSU controls the individual PSU's status LED.
The power status LED on the front panel is controlled via CPLD.

Reply:
So the CPLD will manage the power status LED on the front panel without software assistance?

Reply:
The front panel power status LED needs to be controlled by software via CPLD.

Reply:
OK. Do you intend to control the power status LED somewhere other than here?

Reply:
Currently, the front panel power status LED is not controlled here. If required we can take this as separate work-item.